src/vs/workbench/contrib/debug/browser/variablesView.ts

```diff
@@ -540,10 +545,18 @@ export class VariablesRenderer extends AbstractExpressionsRenderer {
 	}
 
 	protected renderExpression(expression: IExpression, data: IExpressionTemplateData, highlights: IHighlight[]): void {
-		renderVariable(data.elementDisposable, this.commandService, this.hoverService, expression as Variable, data, true, highlights, this.linkDetector);
+		renderVariable(data.elementDisposable, this.commandService, this.hoverService, expression as Variable, data, true, highlights, this.linkDetector, this.displayType);
 	}
 
 	public override renderElement(node: ITreeNode<IExpression, FuzzyScore>, index: number, data: IExpressionTemplateData): void {
+		data.elementDisposable.clear();
+		data.currentElement = node.element;
+		data.elementDisposable.add(this.configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration('debug.showVariableTypes')) {
+				this.displayType = this.configurationService.getValue<IDebugConfiguration>('debug').showVariableTypes;
```

Comment:
I would probably read the configuration during the render call instead of keeping a boolean around. At the moment you have the edge case of the `displayType` boolean not being updated if the configuration is changed while no elements are rendered (i.e. when not paused in the debugger)